packages/amplication-code-gen-types/src/plugins-types.ts

```diff
@@ -0,0 +1,82 @@
+import { AppInfo, DTOs, Entity, Module, Role } from "./code-gen-types";
+
+export interface DsgContext {
+  appInfo: AppInfo;
+  entities: Entity[];
+  roles: Role[];
+  modules: Module[];
+  DTOs: DTOs;
+  plugins: PluginMap;
+}
+
+export type PluginWrapper = (args: any[], func: () => void) => any;
+
+export interface EventParams {}
+
+export interface DsgPlugin {
+  id?: string;
+  name?: string;
+  description?: string;
+  packageName: string;
+}
+
+export type PluginMap = {
+  [K in EventsName]?: {
+    before: (<T>(context: DsgContext, params: any) => T)[];
+    after: (<T>(context: DsgContext, params: any) => T)[];
+  };
+};
+
+export interface CreateServiceModulesParams extends EventParams {
+  before: {
+    entityName: string;
+    entityType: string;
+    entity: Entity;
+    srcDirectory: string;
+    extraMapping: { [key: string]: any };
+  };
+  after: Module[];
+}
+
+export interface CreateControllerModulesParams extends EventParams {
+  before: {
+    resource: string;
+    entityName: string;
+    entityType: string;
+    entityServiceModule: string;
+    entity: Entity;
+    srcDirectory: string;
+    extraMapping: { [key: string]: any };
+  };
+  after: Module[];
+}
+
+export type EventsName = "createServiceModules" | "createControllerModules";
```

Comment:
Can you please try using enum and see if things still work well for you?

Anyway, this should probably be named `EventName`